src/sage/modular/arithgroup/congroup_gammaH.py

```diff
@@ -1171,13 +1177,16 @@ def nregcusps(self):
         N = self.level()
         H = self._list_of_elements_in_H()
 
-        c = ZZ(0)
-        for d in (d for d in divisors(N) if d**2 <= N):
+        c = ZZ.zero()
+        for d in divisors(N):
+            d2 = d**2
+            if d2 > N:
+                break
```

Comment:
```suggestion
                continue
```
You also need to make the same change here I believe.